nixos/modules/services/networking/consul.nix

```diff
@@ -148,107 +107,61 @@ in
           type = types.bool;
         };
       };
-
     };
-
   };
 
   config = mkIf cfg.enable (
-    mkMerge [{
-
-      users.users.consul = {
-        description = "Consul agent daemon user";
-        uid = config.ids.uids.consul;
-        # The shell is needed for health checks
-        shell = "/run/current-system/sw/bin/bash";
-      };
-
-      environment = {
-        etc."consul.json".text = builtins.toJSON configOptions;
-        # We need consul.d to exist for consul to start
-        etc."consul.d/dummy.json".text = "{ }";
-        systemPackages = [ cfg.package ];
-      };
-
-      systemd.services.consul = {
-        wantedBy = [ "multi-user.target" ];
-        after = [ "network.target" ] ++ systemdDevices;
-        bindsTo = systemdDevices;
-        restartTriggers = [ config.environment.etc."consul.json".source ]
-          ++ mapAttrsToList (_: d: d.source)
-            (filterAttrs (n: _: hasPrefix "consul.d/" n) config.environment.etc);
```

Comment:
`Significant enough for changelog entry` (the no-longer-using of `consul.d`)

In particular, this changes how users of the module need to write modular configs.

For example, a common way how we would define additional checks so far was to create more `consul.d` JSON files, e.g. like so:

```nix
{
  systemd.services.consul.restartTriggers = [
    config.environment.etc."consul.d/nginx-service.json".source
  ];

  environment.etc."consul.d/nginx-service.json" = {
    text = builtins.toJSON {
      service = {
        name = "nginx";
        checks = [
          {
            id = "nginx-status-page";
            name = "nginx status page";
            http = "http://localhost:80/nginx_status";
            interval = "1s";
            timeout = "1s";
          }
        ];
      };
    };
  };
}
```

I think we need to explain how this is to be translated to the new setup.

I guess there are 2 ways to do it:

1. `cfg.settingsFiles`
2. translating them to `checks = [ ]` as described in https://www.consul.io/docs/discovery/checks#multiple-check-definitions and putting them into `cfg.settings`.

Does one of these have more benefits / should we recommend one over the other?

For the second approach, does `type = format.type;` take care to concatenate lists if set from multiple different NixOS modules?